### PR TITLE
#19 documentation vignette ADTPET and ADAPET

### PR DESCRIPTION
**Pull request by Fanny-Gautier**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by Fanny-Gautier** [comment]
@pharmaverse/admiralneuro The vignette is almost ready.
Only the section on _Criterion Flags for Amyloid Categories_ remains to be completed, which will be done once issue #27 is completed.
In the meantime, please feel free to review the current version of the vignette and let me know your comments.
Many thanks!

File: vignettes/adpet.Rmd [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+While loading our input data, we can combine the `NV` domain and the `SUPPNV` supplementary domain for easier use in the next steps. Using the `metatools::combine_supp()` function avoids the need to manually transpose and merge the supplementary dataset with the corresponding domain.
+
+```{r, message=FALSE, warning=FALSE}
+nv_suppnv <- combine_supp(nv, suppnv)
+```
+
+## Initial PET Scan Analysis Dataset Set-up {#adpet_setup}
+
+The following steps are used to merge `ADSL` variables with the source data `NV` (combined to `SUPPNV`),
```
Comment (manciniedoardo) [inline vignettes/adpet.Rmd:90]:
```suggestion
The following steps are used to merge `ADSL` variables with the source data `NV` (previously combined with `SUPPNV`),
```

File: vignettes/adpet.Rmd [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+  comment = "#>"
+)
+
+library(admiraldev)
+```
+
+# Introduction
+
+This article describes the creation of PET Scan ADaMs:
+starting with `ADTPET` (Tau PET Scan Analysis Dataset), followed by `ADAPET` (Amyloid PET Scan Analysis Dataset).
```
Comment (manciniedoardo) [inline vignettes/adpet.Rmd:23]: Add some details here to explicitly say that we recommend splitting out Amyloid and Tau into two different ADaMs but the vignette is joint between the two for initial setup, parameter mapping and finals steps, and then there are separate sections where the two diverge.

File: vignettes/adpet.Rmd [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+    )
+  )
+```
+
+Having established the foundation of a global PET Scan Analysis Dataset (`adpet`),
+we can now differentiate between Tau and Amyloid data.
+
+## `ADTPET` (Tau PET Scan Analysis Dataset){#adtpet}
+
+Tau PET scan data do not require any further derivation.
```
Comment (manciniedoardo) [inline vignettes/adpet.Rmd:178]:
```suggestion
Tau PET scan data generally do not require any further derivations.
```

File: vignettes/adpet.Rmd [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+```{r, message=FALSE, warning=FALSE}
+adtpet <- adpet %>%
+  filter(AGCAT == "TAU TRACER")
+```
+
+## `ADAPET` (Amyloid PET Scan Analysis Dataset){#adapet}
+
+From this point forward, this section will focus exclusively on Amyloid data.
+To that end, we will select the Amyloid tracer records from the global PET Scan Analysis Dataset (`adpet`),
+for which the parameter-level and analysis value variables have already been created.
```
Comment (manciniedoardo) [inline vignettes/adpet.Rmd:190]: I don't think we need as many words here, what about:

```suggestion
If producing an `ADAPET`, we can now filter the global PET Scan Analysis Dataset `adpet` to retain only the Amyloid tracer records.
```

File: vignettes/adpet.Rmd [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+  )
+```
+
+Having established the foundation of a global PET Scan Analysis Dataset (`adpet`),
+we can now differentiate between Tau and Amyloid data.
+
+## `ADTPET` (Tau PET Scan Analysis Dataset){#adtpet}
+
+Tau PET scan data do not require any further derivation.
+Therefore, only a selection of the corresponding Tau tracer records is necessary.
```
Comment (manciniedoardo) [inline vignettes/adpet.Rmd:179]:
```suggestion
Therefore, if producing an `ADTPET`, we can now filter the global PET Scan Analysis Dataset `adpet` to retain only the Tau tracer records.
```

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Thanks @Fanny-Gautier - structure looks good, just left a few comments.

File: vignettes/adpet.Rmd [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+## Read in Data {#readdata}
+
+To start, all data frames needed for the creation of `ADTPET` or `ADAPET` should be
+loaded into the global environment. Reading data will usually be a
+company specific process, however, for the purpose of this vignette, we
+will use example data from `{admiralneuro}`. We will
+utilize `NV`, `SUPPNV`, `AG` and `ADSL` for the basis of the Pet Scan ADaM datasets.
+
+```{r, message=FALSE, warning=FALSE}
+nv_neuro <- admiralneuro::nv_neuro
```
[the remark this answers is not in the document]
Comment (Fanny-Gautier) [reply vignettes/adpet.Rmd]: Updated

File: vignettes/adpet.Rmd (hunk 2 above)
Comment (Fanny-Gautier) [reply vignettes/adpet.Rmd:27]: updated

**Comment by Fanny-Gautier** [comment]
>  image

In today's meeting it has been discussed that the `NVMETHOD` and `REFREG` variables must be used as key variables to merge the `lookup_param` table to the derived data. 
`NVMETHOD` and `REFREG` variable must thus be added back in the `lookup_param` of the vignette and added in the `by_vars` argument of the merge statement.

**Comment by Fanny-Gautier** [comment]
Hi @Lina2689 ,
The templates being updated, do you want to add the following parameters in the [vignette](vignettes/adpet.Rmd) and adapt the description txt accordingly?

Still within `slice_derivation()`function as the 2 other parameters (`"CLBFBB"`, `"CLAFBP"`), please add:
1. PARAMCD = "CLBFBP", PARAM = "Centiloid (CL) based on BERKELEY FBP"
2. PARAMCD = "CLAFBB", PARAM = "Centiloid (CL) based on AVID FBB",

The criterion flags (#criterionfl) should also include these 2 new parameters.

**Comment by Fanny-Gautier** [comment]
Hi @Lina2689 ,
The templates being updated, do you want to add the following parameters in the [vignette](vignettes/adpet.Rmd) and adapt the description txt accordingly?

Still within `slice_derivation()`function as the 2 other parameters (`"CLBFBB"`, `"CLAFBP"`), please add:
1. PARAMCD = "CLBFBP", PARAM = "Centiloid (CL) based on BERKELEY FBP"
2. PARAMCD = "CLAFBB", PARAM = "Centiloid (CL) based on AVID FBB",

The criterion flags (#criterionfl) should also include these 2 new parameters.

**Comment by Fanny-Gautier** [comment]
Example of changes in the description text: You could also change the order of a/b/c/d and the code if you want to add back/group by "Florbetaben (FBB)" and "Florbetapir (FBP)".
image


For our examples, we divide the process into two main steps:
[...]

The second step involves converting the original `SUVR` parameters to Centiloid using `admiralneuro::compute_centiloid()`.
To do so, we apply filtering conditions using `admiral::derivation_slice()`.
a. Based on the `SUVRBFBB` tracer, the `BERKELEY FBB SUVR PIPELINE` method, and the `Whole Cerebellum` reference region.
b. Based on the `SUVRAFBP` tracer, the `AVID FBP SUVR PIPELINE` method, and the same reference region (`Whole Cerebellum`).
c. _add the new parameter_
d. _add the 2nd new parameter_

The new parameters values which include `AVAL`, `PARAMCD`, `PARAM`, and `AVALU`,
are assigned within the `args` argument of this function.

**Comment by Lina2689** [comment]
@Fanny-Gautier I am ready with my updates in vignette but I don't see the newly added parameter in `nv_neuro` dataset. Do we still need to add them?
PARAMCD = "CLBFBP", PARAM = "Centiloid (CL) based on BERKELEY FBP"
PARAMCD = "CLAFBB", PARAM = "Centiloid (CL) based on AVID FBB"

Let me know if I am missing something.

**Review by Fanny-Gautier — COMMENTED** [review]
Looks very good! Thanks for the updates Lina!

File: vignettes/adpet.Rmd [hunk 7]
```diff
@@ -0,0 +1,374 @@
...
+)
+
+library(admiraldev)
+```
+
+# Introduction
+
+This article describes the creation of PET Scan ADaMs.
+
+We recommend splitting out Amyloid and Tau into two distinct ADaMs datasets. The vignette presents
```
Comment (jwang-lilly) [inline vignettes/adpet.Rmd:24]: We use lower case "amyloid" and "tau" in sentences, unless they are at the beginning of a sentence.

File: vignettes/adpet.Rmd [hunk 8]
```diff
@@ -0,0 +1,374 @@
...
+    AVALC = if_else(
+      is.na(NVSTRESN) | as.character(NVSTRESN) != NVSTRESC,
+      NVSTRESC,
+      NA
+    )
+  )
+```
+
+Having established the foundation of a global PET Scan Analysis Dataset (`adpet`),
+we can now differentiate between Tau and Amyloid data.
```
Comment (jwang-lilly) [inline vignettes/adpet.Rmd:183]: same here, tau and amyloid data.

File: vignettes/adpet.Rmd [hunk 9]
```diff
@@ -0,0 +1,374 @@
...
+Therefore, if producing an `ADTPET`, we can now filter the global PET Scan Analysis Dataset `adpet` to retain only the Tau tracer records.
+
+```{r, message=FALSE, warning=FALSE}
+adtpet <- adpet %>%
+  filter(AGCAT == "TAU TRACER")
+```
+
+## `ADAPET` (Amyloid PET Scan Analysis Dataset){#adapet}
+
+If producing an `ADAPET`, we can now filter the global PET Scan Analysis Dataset `adpet` to retain only the Amyloid tracer records.
```
Comment (jwang-lilly) [inline vignettes/adpet.Rmd:197]: amyloid tracer

File: vignettes/adpet.Rmd [hunk 10]
```diff
@@ -0,0 +1,374 @@
...
+This is how the parameters and newly converted parameters will look like:
+
+```{r, eval=TRUE, echo=FALSE}
+adapet_param <- distinct(adapet %>% filter(NVTESTCD == "SUVR" | !is.na(AVALU)), USUBJID, PARAMCD, PARAM, AVAL, AVALU, ADT, ADY, VISIT)
+dataset_vignette(adapet_param %>% arrange(USUBJID, ADT), display_vars = exprs(USUBJID, PARAMCD, PARAM, AVAL, AVALU, ADT, ADY, VISIT))
+```
+
+### Derive Criterion Flag Variables for Amyloid Categories{#criterionfl}
+
+Having converted the Standardized Uptake Value Ratio to Centiloid, we can now derive
```
Comment (jwang-lilly) [inline vignettes/adpet.Rmd:334]: maybe use SUVR here as it is defined earlier.

**Comment by Lina2689** [comment]
@manciniedoardo Thanks for catching this issue. 

@jwang-lilly @Fanny-Gautier Please note, one record needs to be added in all the lookup tables which includes templates and vignette. Please see the details below;

~NVTESTCD, ~NVCAT, ~NVLOC, ~REFREG, ~NVMETHOD, ~PARAMCD, ~PARAM, ~PARAMN,
  "SUVR", "**FTP**", "NEOCORTICAL COMPOSITE", "Whole Cerebellum", "AVID **FTP** SUVR PIPELINE", "SUVRA**FTP**", "AVID **FTP** Standard Uptake Ratio Neocortical Composite Whole Cerebellum", **xx**,

**Comment by jwang-lilly** [comment]
Great catch on the missing FTP AVID in lookup.  Added to templates in [issue58](https://github.com/pharmaverse/admiralneuro/pull/58#issue-3297009862)

**Comment by jwang-lilly** [comment]
Perfect!

**Comment by jwang-lilly** [comment]
Copilot is pretty impressive!

**Review by Lina2689 — APPROVED** [review]

**Review by manciniedoardo — APPROVED** [review]
